woocommerce/woocommerce-android
# Order Fulfillment: Add store country at the beginning of the list of shipment providers

PR 2 of 4. Fixes #1058. This PR adds logic to display the list of shipment providers corresponding to the store country first, at the top of the list.

This would be result for a HK store (there are no tracking providers for it):
<img src="https://user-images.githubusercontent.com/22608780/57843861-9a867100-77ec-11e9-83b9-115a0e01363b.png" width="300"/>


This would be result for an Indian store (there is 1 tracking providers for it)
<img src="https://user-images.githubusercontent.com/22608780/57843817-89d5fb00-77ec-11e9-82b9-687bb3c55546.png" width="300"/>

And this would be the result for a US-based store:
<img src="https://user-images.githubusercontent.com/22608780/57843940-bb4ec680-77ec-11e9-9c5d-cf5bda1b6ff8.png" width="300"/>

### Notes:
* The [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1250) must be merged before this can be merged.
* This [PR](https://github.com/woocommerce/woocommerce-android/pull/1054) should be merged first

### Testing
* Navigate to an order > `Fulfill order` > `Add tracking`
* Check that the providers that correspond to the store country show up at the top of the list
* Check that, if there are no providers corresponding to the store country, the list renders alphabetically.
* Check that any provider can be selected, and shipment tracking can be created.
